sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -484,7 +494,10 @@ case class Aggregate(
       }.nonEmpty
     )
 
-    !expressions.exists(!_.resolved) && childrenResolved && !hasWindowExpressions
+    !expressions.exists(!_.resolved) &&
+      childrenResolved &&
+      !hasWindowExpressions &&
+      (isGrouped || groupingExpressions.isEmpty)
```

Comment:
It does not matter if `isGrouped` has any grouping expressions (literal grouping expressions are eliminated during optimization). It is however problematic when a not-grouped `Aggregate` has grouping expressions; this means that we have not derived the `isGrouped` flag  correctly.